src/main/java/dk/kb/discover/api/v1/impl/DsDiscoverApiServiceImpl.java

```diff
@@ -167,12 +176,51 @@ public String solrSearch(String collection, String q, List<String> fq, Integer r
             SolrService solr = SolrManager.getSolrService(collection);
             // TODO: Pass the map of request parameters instead of all parameters as first class
             httpServletResponse.setContentType(solr.getResponseMIMEType(wt)); // Needed by SolrJ
+           
+             //Add filter query from license module.
+            DsLicenseApi licenseClient = getDsLicenseApiClient();
+            GetUserQueryInputDto licenseQueryDto = getLicenseQueryDto();
+            GetUsersFilterQueryOutputDto filterQuery = licenseClient.getUserLicenseQuery(licenseQueryDto);
+            log.info("filter query from licensemodule:"+filterQuery.getFilterQuery()); // in the start this is a very usefull log!            
+            fq.add(filterQuery.getFilterQuery()); //Add the additional filter query
+                        
             return solr.query(q, fq, rows, start, fl, facet, facetField, qOp, wt, version, indent, debug, debugExplainStructured);
         } catch (Exception e){
             throw handleException(e);
         }
     }
 
+    
+    private static GetUserQueryInputDto getLicenseQueryDto() {
+    	   GetUserQueryInputDto getQueryDto = new GetUserQueryInputDto(); 
+          
+    	   getQueryDto.setPresentationType("Search"); // Important. Must be defined in Licensemodule with same name           
+         
+    	   //"everybody=true" is a value everyone will (from keycloak?)
+    	   UserObjAttributeDto everybodyUserAttribute=new UserObjAttributeDto();         
+           everybodyUserAttribute.setAttribute("everybody"); 
+           ArrayList<String> values = new ArrayList<String>();
+           values.add("yes");
+           everybodyUserAttribute.setValues(values);
+           
+           List<UserObjAttributeDto> allAttributes = new ArrayList<UserObjAttributeDto>(); 
+           allAttributes.add(everybodyUserAttribute);
+           
+           getQueryDto.setAttributes(allAttributes);           
+           return getQueryDto;
+    	
+    	
+    }
+    
+    private static DsLicenseApi getDsLicenseApiClient() {
```

Comment:
A new DsLicenseClient will be created for every call to `solrSearch`. Could you make it a Singleton?

Reply:
Done. Hoping it is thread safe :)

Reply:
It is. But thanks for the sanity check.